### ISSUE TITLE
R-Vine matrix order

### ISSUE THREAD

**Issue by tnagler** [tvatter/vinecopulib#48]
@tvatter 
So I finally found a "bug" i've been searching for two days in the structure selection code. The bug is that 
I tried to read the R-vine matrix in a way that makes sense, and not the way VineCopula wants me to do it ;) That reminded me of a other things I didn't like about the way the matrix is oriented. Let me explain with a small example:
```
4 0 0 0
3 3 0 0 
2 2 2 0
1 1 1 1
```
1. Intuitively I read the edge in the last tree as `4, 3; 2 1`, but in VineCopula it's supposed to be read as `3, 4; 2 1`. 
2. One would expect that the `(i, j)` entry of the matrix is related to edge `j` in tree `i`. Instead it relates to edge `j`  in tree `5 - i`. That's also one of the reasons why indexing this matrix in algorithms is so painful (and I reversed the matrices internally). Also it doesn't look like a grapevine ;)

So a solution to both points would be to store the matrix as
```
1 1 1 1
2 2 2 0
3 3 0 0
4 0 0 0
```
and read it from the bottom up, e.g. `4, 3 ; 2 1` etc. 

**So my question is:** Should we use this storage order for the matrix?

The downside would be that we risk to get people confused. There will be three different ways to write this matrix out there (the third is the "upper triangular style" used in Harry Joe's book and library and causes the same indexing problems).

**Comment by tvatter** [comment]
I think that using this rotated matrix for structure storage is an excellent idea. I'm always getting confused by problem 2, and it's a neat solution. However, I don't see how this is related to problem 1 (and I'm sure that most people don't see the difference between `4, 3; 2 1` and `3, 4; 2 1`).

**Comment by tnagler** [comment]
If people don't see the difference between `4, 3; 2 1`  and `3, 4; 2 1`, it's probably because they only think in terms of parametric, exchangeable families with positive association. But relaxing any of these restrictions, the distinction becomes crucial: there is obviously a huge difference between a 90 and 270 degrees rotated Clayton. 

The storage order is related to this problem in the following way: 
- In the old format you start somewhere in the middle, jump to the top and then jump down.
- In the new format you simply read the numbers from the bottom up.

**Comment by tvatter** [comment]
I know that the difference is important, but what I meant is: people don't think too much about such issues unless they are actually implementing the thing or specifying the structure manually.

Concerning the storage order, again I know what you mean and I think that this is an internal issue. We'll need to document it properly, but unless someone contributes to the library or an end-user specifies the structure manually, that's not something people will notice.

Bottom line is: I think that using the order is the way to go. I hate the one from `VineCopula` but I've always been looking at it as some kind of necessary evil. Thus, if there is a better alternative, then we should definitely use it.
